File: 682-baseball-game/solution.py
class Solution(object):
    def calPoints(self, ops):
        """
        :type ops: List[str]
        :rtype: int
        """
        stack, sum_v = [], 0
        for v in ops:
            if v != 'C' and v != 'D' and v != '+':
                stack.append(int(v))
                sum_v += stack[-1]
            elif v == 'C':
                sum_v -= stack[-1]
                stack.pop()
            elif v == 'D':
                stack.append(stack[-1] * 2)
                sum_v += stack[-1]
            elif v == '+':
                stack.append(sum(stack[-2:]))
                sum_v += stack[-1]
                
        return sum_v

File: 682-baseball-game/solution.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        stack<int> my_stack;
        int sum = 0;
        for (int i = 0; i < ops.size(); i ++) {
            if (ops[i].compare("C") !=0 && ops[i].compare("D") !=0 && ops[i].compare("+") != 0) {
                my_stack.push(atoi(ops[i].c_str()));
                sum += my_stack.top();
            } else if (ops[i].compare("C") == 0) {
                sum -= my_stack.top();
                my_stack.pop();
            } else if (ops[i].compare("D") == 0) {
                my_stack.push(my_stack.top()*2);
                sum += my_stack.top();
            } else if (ops[i].compare("+") == 0) {
                int top_1 = my_stack.top();
                my_stack.pop();
                int top_2 = my_stack.top();
                my_stack.pop();
                my_stack.push(top_2);
                my_stack.push(top_1);
                my_stack.push(top_1+top_2);
                sum += my_stack.top();

            }
        }
        return sum;
    }
};